tests/stub/authorization.py

```diff
@@ -655,3 +655,70 @@ def get_vars(self, host=None):
 
     def get_db(self):
         return None
+
+
+class NoRoutingAuthorizationTests(TestkitTestCase):
+    def setUp(self):
+        super().setUp()
+        self._readServer = StubServer(9010)
+        self._uri = "bolt://%s:%d" % (self._readServer.host,
+                                      self._readServer.port)
+        self._auth = types.AuthorizationToken(
+            scheme="basic", principal="p", credentials="c")
+        self._userAgent = "007"
+
+    def tearDown(self):
+        self._readServer.reset()
+        super().tearDown()
+
+    def read_return_1_failure_return_2_succeed_script(self):
+        return """
+        !: BOLT 4
+        !: AUTO HELLO
+        !: AUTO GOODBYE
+        !: AUTO RESET
+        !: ALLOW RESTART
+        !: ALLOW CONCURRENT
```

Comment:
`!: ALLOW CONCURRENT` implies `!: ALLOW RESTART`. No need but also no harm in having it there explicitly.